public/app/features/dashboard/components/AddLibraryPanelWidget/AddLibraryPanelWidget.tsx

```diff
@@ -0,0 +1,114 @@
+import { css, cx, keyframes } from '@emotion/css';
+import React from 'react';
+import { connect, MapDispatchToProps } from 'react-redux';
+import tinycolor from 'tinycolor2';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { LibraryPanel } from '@grafana/schema';
+import { IconButton, useStyles2 } from '@grafana/ui';
+import { addPanel } from 'app/features/dashboard/state/reducers';
+
+import {
+  LibraryPanelsSearch,
+  LibraryPanelsSearchVariant,
+} from '../../../library-panels/components/LibraryPanelsSearch/LibraryPanelsSearch';
+import { DashboardModel, PanelModel } from '../../state';
+
+export type PanelPluginInfo = { id: number; defaults: { gridPos: { w: number; h: number }; title: string } };
+
+export interface OwnProps {
+  panel: PanelModel;
+  dashboard: DashboardModel;
+}
+
+export interface DispatchProps {
+  addPanel: typeof addPanel;
+}
+
+export type Props = OwnProps & DispatchProps;
+
+export const AddLibraryPanelWidgetUnconnected = ({ panel, dashboard }: Props) => {
+  const onCancelAddPanel = (evt: React.MouseEvent<HTMLButtonElement>) => {
+    evt.preventDefault();
+    dashboard.removePanel(panel);
+  };
+
+  const onAddLibraryPanel = (panelInfo: LibraryPanel) => {
+    const { gridPos } = panel;
+
+    const newPanel = {
+      ...panelInfo.model,
+      gridPos,
+      libraryPanel: panelInfo,
+    };
+
+    dashboard.addPanel(newPanel);
+    dashboard.removePanel(panel);
+  };
+
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={cx('panel-container', styles.callToAction)}>
+        <div className={cx(styles.headerRow, 'grid-drag-handle')}>
+          <span>Add panel from panel library</span>
+          <div className="flex-grow-1" />
+          <IconButton aria-label="Close 'Add Panel' widget" name="times" onClick={onCancelAddPanel} />
+        </div>
+        <LibraryPanelsSearch onClick={onAddLibraryPanel} variant={LibraryPanelsSearchVariant.Tight} showPanelFilter />
+      </div>
+    </div>
+  );
+};
+
+const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = { addPanel };
+
+export const AddLibraryPanelWidget = connect(undefined, mapDispatchToProps)(AddLibraryPanelWidgetUnconnected);
```

Comment:
isn't it an old way of wrapping a component with redux? 
I think would be good to use hooks

Reply:
so, we have useSelector and useDispatch for this